internal/service/certificatesigningrequest.go

```diff
@@ -272,33 +271,45 @@ func (h *ServiceHandler) ApproveCertificateSigningRequest(ctx context.Context, r
 	if err != nil {
 		if errors.Is(err, flterrors.ErrResourceNotFound) {
 			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
-		} else {
-			return nil, err
 		}
+		return nil, err
 	}
 
 	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestDenied) {
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been denied"}, nil
 	}
 
-	approvedCondition := api.Condition{
-		Type:    api.CertificateSigningRequestApproved,
-		Status:  api.ConditionStatusTrue,
-		Reason:  "Approved",
-		Message: "Approved",
-	}
-	err = h.store.CertificateSigningRequest().UpdateConditions(ctx, orgId, request.Name, []api.Condition{approvedCondition})
+	signedCert, err := signApprovedCertificateSigningRequest(h.ca, csr)
 	if err != nil {
 		if errors.Is(err, flterrors.ErrResourceNotFound) {
 			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
 		}
+		if errors.Is(err, flterrors.ErrResourceVersionConflict) {
+			return server.ApproveCertificateSigningRequest409JSONResponse{}, nil
+		}
 		return nil, err
 	}
 
-	err = signApprovedCertificateSigningRequest(h.ca, csr)
+	err = h.store.CertificateSigningRequest().UpdateCertificate(ctx, orgId, request.Name, signedCert)
```

Comment:
Should not from this point everything be part of transaction? I mean in case https://github.com/flightctl/flightctl/pull/491/files#diff-d5e3988a9944d806d028372b1fc8f031511b4f6495c386eebf33bca63f003e6eR309 will fail should not all the updates done previously be reverted?

Reply:
Updating the certificate here is what make the operation meaningful - and if this fails, we return an error and don't set condition to "approved". If the certificate has been added but then setting the condition fails (below), we could remove the certificate before returning an error - I can make this update if it seems better.

In the future we will collect _n_ number of approvals (_n_ being set by policy) before issuing the cert, so these will get separated out a bit more than they are here.

Reply:
I've changed it to remove the certificate that was previously set, in the case that approval fails.

Reply:
I believe that UpdateCertificate and UpdateConditions should be atomic db operation, right? 
I mean we should not create and than delete but we should rather do both in the same db call.

Reply:
Currently it is (or should be) atomic to be correct, but they will be separate later.
I can make UpdateCertificate and UpdateConditions into one function and have a todo to separate them out later.

Reply:
UpdateCertificate can just set "approved" to "true".